### DISCUSSION TITLE
Error: MongoDB error Command failed (DuplicateKey): Index build failed:

### DISCUSSION THREAD

**Discussion by Redskinsjo** [prisma/prisma#16691] [Q&A] [answered]
Hi,

I am pushing a new schema, after having modified a key, and now I can't finalise pushing new schema because old key seems to not be erased, even if I use flag --accept-data-loss.

```
Error: MongoDB error
Command failed (DuplicateKey): Index build failed: _ID_: Collection _database_.jobcontexts ( _ID_ ) :: caused by :: E11000 duplicate key error collection: _database_.jobcontexts index: jobcontexts_jobId_key dup key: { jobId: null })
   0: migration_core::state::SchemaPush
             at migration-engine/core/src/state.rs:381
```

```
model contexts {
  id     String           @id @default(auto()) @map("_id") @db.ObjectId
  v      Int              @map("__v")
  family String
  titles jobcontexttitles
  job    jobs      @relation(fields: [jobId], references: [id])
  jobId  String           @unique @db.ObjectId
}

model jobs {
  id           String               @id @default(auto()) @map("_id") @db.ObjectId
  v            Int                  @map("__v")
  href         String
  title        String
  context      contexts?
  competencies competencies[]
  panel        panels         @relation(fields: [panelId], references: [id])
  panelId      String               @db.ObjectId
}
```

I understand that prisma migrate dev should not work with mongodb, but prisma db push is the command I am using (and that I should use?)

**Comment by Redskinsjo** [comment, marked as answer]
solved it by setting 'context    contexts[]' in:
```
model jobs {
  id           String               @id @default(auto()) @map("_id") @db.ObjectId
  v            Int                  @map("__v")
  href         String
  title        String
  context      contexts[]
  competencies competencies[]
  panel        panels         @relation(fields: [panelId], references: [id])
  panelId      String               @db.ObjectId
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
